Dataset loading

abdirahmanbm01 opened the pull request:
Adds a dataset loader that normalizes reasoning datasets to a common {input, target, meta} format.

Features:
- Single load_normalized() function to load any configured dataset
- Validates that expected splits exist
- Supports HuggingFace datasets and local JSONL files
- Config-based mapping in utils/config.py

Supported datasets: gsm8k, math, hotpotqa, strategyqa, piqa, commonsenseqa, mbpp, humaneval, aime24

Usage:
from datasets import load_normalized
data = load_normalized("gsm8k")
train = data["train"]  # examples have {input, target, meta} format

Files: datasets/loader.py, datasets/__init__.py, datasets/smoke_test.py, utils/config.py, requirements.txt

shavonthadani committed:
Add dataset loader with normalization to {input, target, meta} format

shavonthadani committed:
Merge branch 'dataset-loading' of https://github.com/shavonthadani/Mira into dataset-loading

jerryfrancis-97 reviewed (changes requested):
<img width="1638" height="334" alt="image" src="https://github.com/user-attachments/assets/4fd0cd79-b9db-46f3-b496-67d018cc8bbf" />

@abdirahmanbm01 wasn't able to run both the smoke test and the sample code (usage) provided above. "datasets" name usage might be the issue, being externally imported and locally created as different libs with same name.

Can you please check and lmk?

shavonthadani committed:
Fix dataset loader: rename package to mira_datasets and update all HF dataset IDs

- Rename datasets/ to mira_datasets/ to avoid naming conflict with HuggingFace datasets library
- Update all dataset HF IDs to correct versions:
  - gsm8k: openai/gsm8k
  - math: HuggingFaceH4/MATH-500
  - hotpotqa: hotpotqa/hotpot_qa
  - strategyqa: ChilleD/StrategyQA
  - piqa: nthngdy/piqa
  - commonsenseqa: tau/commonsense_qa
  - aime24: HuggingFaceH4/aime_2024
  - zebralogic: WildEval/ZebraLogic (new)
- Fix split configurations for datasets
- Replace Unicode arrow character in smoke_test.py
- 9/10 datasets now working successfully

jerryfrancis-97 reviewed (approved):
Smoke test works - python -m mira_datasets.smoke_test , 
Usage also works.

LGTM, approved
